src/Features/CSharp/Portable/ConvertToRawString/ConvertRegularStringToRawStringCodeRefactoringProvider.cs

```diff
@@ -128,20 +164,148 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             SyntaxFormattingOptions options,
             CancellationToken cancellationToken)
         {
-            return kind switch
+            var characters = CSharpVirtualCharService.Instance.TryConvertToVirtualChars(token);
+            Contract.ThrowIfTrue(characters.IsDefaultOrEmpty);
+
+            // If the user asked to remove whitespace then do so now.
+            if (kind == ConvertToRawKind.MultiLineWithoutLeadingWhitespace)
+                characters = CleanupWhitespace(characters);
+
+            return kind == ConvertToRawKind.SingleLine
+                ? ConvertToSingleLineRawString(token, characters)
+                : ConvertToMultiLineRawIndentedString(document, token, options, characters, cancellationToken);
+        }
+
+        private static VirtualCharSequence CleanupWhitespace(VirtualCharSequence characters)
+        {
+            using var _ = ArrayBuilder<VirtualCharSequence>.GetInstance(out var lines);
+
+            // First, determine all the lines in the content.
+            BreakIntoLines(characters, lines);
+
+            // Remove the leading and trailing line if they are all whitespace.
+            while (lines.Count > 0 && AllWhitespace(lines.First()))
+                lines.RemoveAt(0);
+
+            while (lines.Count > 0 && AllWhitespace(lines.Last()))
+                lines.RemoveAt(lines.Count - 1);
+
+            if (lines.Count == 0)
+                return VirtualCharSequence.Empty;
+
+            // Use the remaining lines to figure out what common whitespace we have.
+            var commonWhitespacePrefix = ComputeCommonWhitespacePrefix(lines);
+
+            var result = ImmutableSegmentedList.CreateBuilder<VirtualChar>();
+
+            foreach (var line in lines)
             {
-                ConvertToRawKind.SingleLine => ConvertToSingleLineRawString(token),
-                ConvertToRawKind.MultiLine => ConvertToMultiLineRawString(token, options.NewLine),
-                ConvertToRawKind.MultiLineIndented => ConvertToMultiLineRawIndentedString(document, token, options, cancellationToken),
-                _ => throw ExceptionUtilities.UnexpectedValue(kind),
-            };
+                if (AllWhitespace(line))
+                {
+                    // For an all-whitespace line, just add the trailing newlines on the line (if present).
+                    AddRange(result, line.SkipWhile(IsCSharpWhitespace));
+                }
+                else
+                {
+                    // Normal line.  Skip the common whitespace.
+                    AddRange(result, line.Skip(commonWhitespacePrefix));
+                }
+            }
+
+            // Remove all trailing whitespace and newlines from the final string.
+            while (result.Count > 0 && (IsCSharpNewLine(result[^1]) || IsCSharpWhitespace(result[^1])))
```

Comment:
it's slightly different.  the first loop is removing entirely blank lines.  The last loop is removing any trailing newline from the last line (which is not an allwhitespace line (since htat would have already been removed)).